src/geneweaver/api/controller/genesets.py

```diff
@@ -51,3 +113,16 @@ def upload_geneset(
     """Upload a geneset."""
     db_geneset_value.format_geneset_values_for_file_insert(geneset.gene_list)
     return {"geneset_id": 0}
+
+
+def get_gene_identifier_type(gene_id_type: int) -> GeneIdentifier:
```

Comment:
This is mainly a small helper method to avoid redundancy on the controller functions. There isn't much business logic in it except to verify that the identifier is valid. It is preferable to keep the HTTP exceptions to the controller layer. Keeping this method in the controller for now.

Reply:
Yeah, that makes sense to me too.